R/oauth-flow-auth-code.R

```diff
@@ -123,20 +125,27 @@ req_oauth_auth_code <- function(req, client,
 #' @param auth_params List containing additional parameters passed to `oauth_flow_auth_code_url()`
 #' @param token_params List containing additional parameters passed to the
 #'   `token_url`.
-#' @param host_name `r lifecycle::badge("deprecated")` Use `redirect_uri`
-#'   instead.
+#' @param host_name,host_ip,port `r lifecycle::badge("deprecated")`
+#'   Now use `redirect_uri` instead.
 #' @param host_ip IP address for the temporary webserver used to capture the
 #'   authorization code.
-#' @param type Either `desktop` or `web`. Use desktop when running on the
-#'   desktop in an environment where you can redirect the user to `localhost`.
-#'   Use `web` when running in a hosted web environment.
-#' @param port Port to bind the temporary webserver to. Used only when
-#'   `redirect_uri` is `"http(s)://localhost"`. By default, this uses a random
-#'   port. You may need to set it to a fixed port if the API requires that the
-#'   `redirect_uri` specified in the client exactly matches the `redirect_uri`
-#'   generated by this function.
+#' @param port Port to bind the temporary webserver to.
 #' @param redirect_uri URL to redirect back to after authorization is complete.
 #'   Often this must be registered with the API in advance.
+#'
+#'   httr2 supports two forms of redirect. Firstly, you can use a `localhost`
+#'   url (the default), where httr2 will set up a temporary webserver to listen
+#'   for the OAuth redirect. In this case, httr2 will automatically append a
+#'   random port. If you need to set it to a fixed port because the API requires
+#'   it, then specify it with (e.g.) `"http://localhost:1011"`. This technique
+#'   works well when you are working on your own computer.
+#'
+#'   Alternatively, you can provide a URL to a website that uses javascript to
+#'   give the user a code to copy and paste back into the R session (see
+#'   <https://www.tidyverse.org/google-callback/>, for an example). This is
```

Comment:
FWIW gargle includes the source for a similar page (but minus the tidyverse branding), in case you want to do the same or point at that one:

https://github.com/r-lib/gargle/blob/main/inst/pseudo-oob/google-callback/index.html

Reply:
I haven't traced through the httr2 code enough yet to know if this is already here: But the gargle example doesn't just emit the naked auth code. It checks for a `state` and then packages/encodes the `state` and `code`. 

https://github.com/r-lib/gargle/blob/ef00eb0581a8eda05185a3a6348f80cff522dd76/inst/pseudo-oob/google-callback/index.html#L46-L49

Which then gets decoded and checked:

https://github.com/r-lib/gargle/blob/ef00eb0581a8eda05185a3a6348f80cff522dd76/R/oauth-init.R#L134-L138

Reply:
Yeah I added support for that in the original PR. It should work.

Reply:
Ah, ok. I was just focusing on the specific diffs in front of my 👀